packages/cw-orch-core/src/contract/paths.rs

```diff
@@ -142,13 +142,33 @@ mod artifacts_dir {
 
         /// Find a WASM file in the artifacts directory that contains the given name.
         pub fn find_wasm_path(&self, name: &str) -> Result<WasmPath, CwEnvError> {
+            self.find_wasm_path_with_build_postfix(name, <BuildPostfix>::None)
+        }
+
+        /// Find a WASM file in the artifacts directory that contains the given contract name AND build post-fix.
+        /// If a build with the post-fix is not found, the default build will be used.
+        /// If none of the two are found, an error is returned.
+        pub fn find_wasm_path_with_build_postfix<T: ChainState>(
+            &self,
+            name: &str,
+            build_postfix: BuildPostfix<T>,
+        ) -> Result<WasmPath, CwEnvError> {
+            let build_postfix: String = build_postfix.into();
             let path_str = fs::read_dir(self.path())?
                 .find_map(|entry| {
                     let path = entry.ok()?.path();
                     let file_name = path.file_name().unwrap_or_default().to_string_lossy();
-                    if path.is_file()
-                        && path.extension().unwrap_or_default() == "wasm"
-                        && file_name.contains(name)
+                    if !path.is_file() {
+                        return None;
+                    }
+
+                    if (path.extension().unwrap_or_default() == "wasm"
```

Comment:
This line is weird, what are the conditions we want here : 
1. We need a wasm file
2. We need it to either be non-postfixed or post-fixed

It looks like the condition here is just is_default_artifact, and that's it, don't understand why you would need the postfix validation here. You are not adding build postfix there in the path anyway

Reply:
It's different because `is_default_artifact` matches on the default name (i.e only the contract name and no postfixes) so it would return false for a build with a postfix.